pkg/internal/apis/config/default.go

```diff
@@ -41,31 +41,39 @@ func SetDefaultsCluster(obj *Cluster) {
 		SetDefaultsNode(a)
 	}
 	if obj.Networking.IPFamily == "" {
-		obj.Networking.IPFamily = "ipv4"
+		obj.Networking.IPFamily = IPv4Family
+	} else if obj.Networking.IPFamily == DualStackFamily {
+		obj.FeatureGates["IPv6DualStack"] = true
```

Comment:
is ok defaulting it here?

Reply:
this isn't going to work if featureGates has gates set but dualstack isn't explicitly set.
also we need to set this gate only for certain k8s versions.

we should do the defaulting further down the stack, and it should be *defaulting*, if the user explicitly sets it to false then 🤷

Reply:
yeah, is bad shortcut

Reply:
> this isn't going to work if featureGates has gates set but dualstack isn't explicitly set.

why? this is a kubeadm feature-gate, not a kubernetes feature gate.

> need to set this gate only for certain k8s versions.

I assume that kubeadm will fail if the gate doesn't exist

Reply:
actually, I think that this is correct, is setting the kubeadm feature gate to true if the IPFamily == DualStack.
If the user sets it to false, he is doing something contradictory, because he is telling kind to deploy a dual-stack cluster and configuring the gate to not do it.
I think that the problem here is the semantics of defaulting, but my logic is if we do this for podSubnets
```
	// default the pod CIDR
	if obj.Networking.PodSubnet == "" {
		obj.Networking.PodSubnet = "10.244.0.0/16"
		if obj.Networking.IPFamily == "ipv6" {
			// node-mask cidr default is /64 so we need a larger subnet, we use /56 following best practices
			// xref: https://www.ripe.net/publications/docs/ripe-690#4--size-of-end-user-prefix-assignment---48---56-or-something-else-
			obj.Networking.PodSubnet = "fd00:10:244::/56"
		}
	}
```

we should be able to do the same for dual-stack IPFamily, just the difference is that it implease setting the kubeadm feature-gate

Reply:
scratch it, I'm more trying to find an excuse, this does not make sense here